### PR TITLE
XML and FedId configuration update for PPS

### PR DESCRIPTION
**Pull request by ChrisMisan**
#### PR description:
This PR updates the xml mappings and FedId configuration for PPS run3. Backward compatibility is sustained.

#### PR validation:
This PR was validated with relvals 136.793, 136.874 (which run PPS reconstruction). New run3 xml's were parsed without issue. We lack the run3 data to conduct more sophisticated tests. For the run2 data, there should be no changes as to the results.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:
no

### CODE REVIEW

**Comment by tvami** [comment]
Hi @ChrisMisan has it been considered to move these into a DB object as well? Having this IOV structure that you have in the config file is exactly what the multi-IOV tags are for.

**Comment by tvami** [comment]
test parameters:
  - workflows = 136.793, 136.874

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by ChrisMisan** [comment]
Both issues with relvals should be resolved shortly:
```Node station not allowed within station block.``` is due to minor error in one of the xmls, it was parsed fine but the hierarchy is missing one tag.

```No configuration for event run: 326479 lumi: 0 event: 0``` it's true - it's outside our run2 interval, maybe interval is actually different then what we have in the config

@tvami To answer your question, I know this has been discussed, I guess @grzanka and @avati can better state the reasons for the current configuration.

**Comment by grzanka** [comment]
> Hi @ChrisMisan has it been considered to move these into a DB object? Having this IOV structure that you have in the config file is exactly what the multi-IOV tags are for.

Hi @tvami . We can consider moving into a DB objects, I agree it is a cleaner solution. But again we are under time pressure before Run3. Migrating to DB would require some significant changes in our RAW2DATA modules, proper testing etc...

**Comment by malbouis** [comment]
please test

**Comment by francescobrivio** [comment]
I see some differences in the results:
 - [136.7611 log](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_3_X_2022-03-05-1100+7b2143/48739/validateJR/all_OldVSNew_RunJetHT2016EreMINIAODwf136p7611/) --> this is known and comes from another PR
 - [139.001 log](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_3_X_2022-03-05-1100+7b2143/48739/validateJR/all_OldVSNew_RunMinimumBias2021wf139p001/) and [138.4 log](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_3_X_2022-03-05-1100+7b2143/48739/validateJR/all_OldVSNew_RunMinimumBias2021PROMPTwf138p4/) --> these are from 2021 PilotBeams wfs, @ChrisMisan could you have a look to see if they are somehow connected to this PR? (although PPS wasn't even installed at that time...)

**Comment by ChrisMisan** [comment]
@francescobrivio I verified that those changes were caused by this PR, is it possible that those wfs are running without ctpps era modifier?

File: EventFilter/CTPPSRawToDigi/python/ctppsDiamondRawToDigi_cfi.py [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
 
+from Configuration.Eras.Modifier_ctpps_2016_cff import ctpps_2016
+from Configuration.Eras.Modifier_ctpps_2017_cff import ctpps_2017
+from Configuration.Eras.Modifier_ctpps_2018_cff import ctpps_2018
+
 from EventFilter.CTPPSRawToDigi.totemVFATRawToDigi_cfi import totemVFATRawToDigi
 
 ctppsDiamondRawToDigi = totemVFATRawToDigi.clone(
     subSystem = cms.string('TimingDiamond'),
+    fedIds = cms.vuint32(579, 581, 582, 583),
```
Comment (jpata) [inline EventFilter/CTPPSRawToDigi/python/ctppsDiamondRawToDigi_cfi.py:11]: might be useful to specify with a comment where these numbers (also in the other files) are coming from.

Comment (vavati) [reply EventFilter/CTPPSRawToDigi/python/ctppsDiamondRawToDigi_cfi.py:11]: Not much to say from where these numbers are coming: these are Fed numbers assigned in the HW, as declared in
DataFormats/FEDRawData/interface/FEDNumbering.h

File: EventFilter/CTPPSRawToDigi/python/ctppsDiamondRawToDigi_cfi.py [hunk 2]
```diff
@@ -1,11 +1,20 @@
 import FWCore.ParameterSet.Config as cms
 
+from Configuration.Eras.Modifier_ctpps_2016_cff import ctpps_2016
+from Configuration.Eras.Modifier_ctpps_2017_cff import ctpps_2017
+from Configuration.Eras.Modifier_ctpps_2018_cff import ctpps_2018
+
 from EventFilter.CTPPSRawToDigi.totemVFATRawToDigi_cfi import totemVFATRawToDigi
 
 ctppsDiamondRawToDigi = totemVFATRawToDigi.clone(
     subSystem = cms.string('TimingDiamond'),
```
Comment (perrotta) [inline EventFilter/CTPPSRawToDigi/python/ctppsDiamondRawToDigi_cfi.py:10]:
```suggestion
    subSystem = 'TimingDiamond',
    fedIds = [579, 581, 582, 583],
    RawToDigi = dict(
        testCRC = 0,                     # no need to test CRC for diamond frames
        testECMostFrequent = 0, # show error in the DQM and then DAQ is sending resync, no need to test in the unpacker
    )
```

File: EventFilter/CTPPSRawToDigi/python/ctppsDiamondRawToDigi_cfi.py [hunk 3]
```diff
@@ -1,11 +1,20 @@
...
+    fedIds = cms.vuint32(579, 581, 582, 583),
     RawToDigi = totemVFATRawToDigi.RawToDigi.clone(
         testCRC = cms.uint32(0), # no need to test CRC for diamond frames
         testECMostFrequent = cms.uint32(0) # show error in the DQM and then DAQ is sending resync, no need to test in the unpacker
     )
 )
+
+# for Run 2 backward compatibility
+(ctpps_2016 | ctpps_2017 | ctpps_2018).toModify(ctppsDiamondRawToDigi,
+fedIds = cms.vuint32())
```
Comment (perrotta) [inline EventFilter/CTPPSRawToDigi/python/ctppsDiamondRawToDigi_cfi.py:20]:
```suggestion
fedIds = [] )
```

File: EventFilter/CTPPSRawToDigi/python/totemRPRawToDigi_cfi.py [hunk 4]
```diff
@@ -1,7 +1,16 @@
...
+from Configuration.Eras.Modifier_ctpps_2016_cff import ctpps_2016
+from Configuration.Eras.Modifier_ctpps_2017_cff import ctpps_2017
+from Configuration.Eras.Modifier_ctpps_2018_cff import ctpps_2018
+
 from EventFilter.CTPPSRawToDigi.totemVFATRawToDigi_cfi import totemVFATRawToDigi
 
 totemRPRawToDigi = totemVFATRawToDigi.clone(
-    subSystem = cms.string('TrackingStrip')
+    subSystem = cms.string('TrackingStrip'),
+    fedIds = cms.vuint32(578, 580, 584, 585)
```
Comment (perrotta) [inline EventFilter/CTPPSRawToDigi/python/totemRPRawToDigi_cfi.py:11]:
```suggestion
    subSystem = 'TrackingStrip',
    fedIds = [578, 580, 584, 585]
```

File: EventFilter/CTPPSRawToDigi/python/totemRPRawToDigi_cfi.py [hunk 5]
```diff
@@ -1,7 +1,16 @@
...
 )
+
+# for Run 2 backward compatibility
+(ctpps_2016 | ctpps_2017 | ctpps_2018).toModify(totemRPRawToDigi,
+fedIds = cms.vuint32())
```
Comment (perrotta) [inline EventFilter/CTPPSRawToDigi/python/totemRPRawToDigi_cfi.py:16]:
```suggestion
fedIds = [] )
```

File: EventFilter/CTPPSRawToDigi/python/totemTimingRawToDigi_cfi.py [hunk 6]
```diff
@@ -1,13 +1,15 @@
 import FWCore.ParameterSet.Config as cms
 
+from Configuration.Eras.Modifier_ctpps_2016_cff import ctpps_2016
+from Configuration.Eras.Modifier_ctpps_2017_cff import ctpps_2017
+from Configuration.Eras.Modifier_ctpps_2018_cff import ctpps_2018
+
 from EventFilter.CTPPSRawToDigi.totemVFATRawToDigi_cfi import totemVFATRawToDigi
 
 totemTimingRawToDigi = totemVFATRawToDigi.clone(
     subSystem = cms.string('TotemTiming'),
```
Comment (perrotta) [inline EventFilter/CTPPSRawToDigi/python/totemTimingRawToDigi_cfi.py:10]:
```suggestion
    subSystem = 'TotemTiming',
```
... and the same in the following

File: EventFilter/CTPPSRawToDigi/python/totemTimingRawToDigi_cfi.py [hunk 7]
```diff
@@ -26,3 +28,7 @@
     printUnknownFrameSummary = cms.untracked.uint32(0)
   )
 )
+
+# for Run 2 backward compatibility
+(ctpps_2016 | ctpps_2017 | ctpps_2018).toModify(totemTimingRawToDigi,
+fedIds = cms.vuint32())
```
Comment (perrotta) [inline EventFilter/CTPPSRawToDigi/python/totemTimingRawToDigi_cfi.py:34]:
```suggestion
fedIds = [] )
```

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by jpata** [comment]
+reconstruction
- PPS fedIds for Run3, corresponding reco changes
- no changes for Run2

**Comment by tvami** [comment]
+alca
 * with the remark that these object might be moved to DB in the future

**Comment by perrotta** [comment]
+1

**Pull request merged**